src/PythonQtConversion.cpp

```diff
@@ -998,8 +999,11 @@ QVariant PythonQtConv::PyObjToQVariant(PyObject* val, int type)
     } else if (PyLong_Check(val)) {
       // return int if the value fits into that range,
       // otherwise it would not be possible to get an int from Python 3
-      qint64 d = PyLong_AsLongLong(val);
-      if (d > std::numeric_limits<int>::max() ||
+      int overflow = 0;
+      qint64 d = PyLong_AsLongLongAndOverflow(val, &overflow);
+      if (overflow != 0) {
+        type = QVariant::ByteArray; // if it's too long for long long return byte array
```

Comment:
Setting the type to QVariant::ByteArray leads to the use of PythonQtConv::PyObjGetBytes later on, which only works if PyBytes_Check returns true. So I guess this doesn't do anything useful.